src/Symfony/Component/JsonPath/Functions/AbstractJsonPathFunction.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonPath\Functions;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+abstract class AbstractJsonPathFunction implements JsonPathFunctionInterface
+{
+    public function __construct(
+        private ?string $name = null,
```

Comment:
This should not use a promoted property, to decouple the parameter type (nullable) from the private property type (not nullable)